src/istio/control/http/service_context.cc

```diff
@@ -72,6 +74,8 @@ void ServiceContext::InjectForwardedAttributes(
     attributes.MergeFrom(service_config_->forward_attributes());
   }
 
+  client_context_->AddForwardAttributes(&attributes);
```

Comment:
This should be the first call in the function. We want overriding happen in finer-grained config, not the other way around.